### PR TITLE
Fix/perf internal search

### PR DESCRIPTION
**Pull request by baptistemesta**

### CODE REVIEW

File: bpm/bonita-core/bonita-process-instance/bonita-process-instance-impl/src/main/java/org/bonitasoft/engine/core/process/instance/event/impl/EventInstanceServiceImpl.java [hunk 1]
```diff
@@ -233,20 +229,13 @@ public void deleteWaitingEvent(final SWaitingEvent waitingEvent) throws SWaiting
...
-                .getFlowNodeNameKey(), OrderByType.ASC);
-        final FilterOption filterOption = new FilterOption(SWaitingEvent.class, BuilderFactory.get(SWaitingMessageEventBuilderFactory.class)
-                .getFlowNodeInstanceIdKey(), flowNodeInstance.getId());
-        final List<FilterOption> filters = Collections.singletonList(filterOption);
-        final QueryOptions queryOptions = new QueryOptions(0, 100, Collections.singletonList(orderByOption), filters, null);
-
-        List<SWaitingEvent> waitingEvents = searchWaitingEvents(SWaitingEvent.class, queryOptions);
+        List<SWaitingEvent> waitingEvents;
         do {
+            waitingEvents = getPersistenceService().selectList(new SelectListDescriptor<SWaitingEvent>("getWaitingEventsOfFlowNode", Collections.<String, Object>singletonMap("flowNodeInstanceId", flowNodeInstance.getId()), SWaitingEvent.class, new QueryOptions(0, 100)));
```
Comment (ermedeiros) [inline bpm/bonita-core/bonita-process-instance/bonita-process-instance-impl/src/main/java/org/bonitasoft/engine/core/process/instance/event/impl/EventInstanceServiceImpl.java]: It's better to use BuilderFactory.get(SWaitingMessageEventBuilderFactory.class).getFlowNodeInstanceIdKey() to retrieve the column name.

Comment (baptistemesta) [inline bpm/bonita-core/bonita-process-instance/bonita-process-instance-impl/src/main/java/org/bonitasoft/engine/core/process/instance/event/impl/EventInstanceServiceImpl.java]: its not the column name here, its the auery parameter

Comment (ermedeiros) [inline bpm/bonita-core/bonita-process-instance/bonita-process-instance-impl/src/main/java/org/bonitasoft/engine/core/process/instance/event/impl/EventInstanceServiceImpl.java]: ok.

**Comment by baptistemesta** [comment]
tests are running but they seems ok! :)

**Pull request merged**